src/Standards/PSR2/Sniffs/Classes/ClassDeclarationSniff.php

```diff
@@ -314,12 +314,14 @@ public function processOpen(File $phpcsFile, $stackPtr)
             if ($checkingImplements === true
                 && $multiLineImplements === true
                 && ($tokens[($className - 1)]['code'] !== T_NS_SEPARATOR
-                || $tokens[($className - 2)]['code'] !== T_STRING)
+                || ($tokens[($className - 2)]['code'] !== T_STRING
+                && $tokens[($className - 2)]['code'] !== T_NAMESPACE))
             ) {
                 $prev = $phpcsFile->findPrevious(
                     [
                         T_NS_SEPARATOR,
                         T_WHITESPACE,
+                        T_NAMESPACE,
```

Comment:
Is this change necessary? The code seems to behave the same way with and without this line.

Reply:
You are right, it is not necessary. I did it for consistency more than anything else. Removing it now.

I will leave this for a proper sniff review as per:

> Note: at a glance, this sniff could probably do with a more thorough review and more defensive token walking/more precise checking, but that's outside the scope of this PR.

For the record: the above note was triggered by seeing code like line 451-453 which checks that the code _after_ a name is not a `\` and not a comma and then presumes it is whitespace, but doesn't actually check if the token is whitespace (and replaces the token anyhow, potentially destroying valid code).

This will need further investigation (and proof of the bug via a code sample, not just my experience telling me there is a likely to be bug). It might be that the code earlier on in the sniff already checks for `\`, comma and whitespace, so that if the next token is not `\` or comma, the sniff can be sure that it will be whitespace (but then again, why not check for that in line 451-453 ?). Investigating that was outside the scope of the current PR though.
Considering the bugs now fixed and seeing that code, I have a feeling there will be more, potentially incorrect, presumptions build into the sniff, which is the reason why I added that note.

Reply:
I've opened ticket #425 to prevent these notes about the sniff review being lost once this PR has been merged.